File: 0283-move-zeroes/0283-move-zeroes.cpp
class Solution {
public:
    void moveZeroes(vector<int>& nums) {
        
        int size = nums.size();
        vector<int>ans(size);
        
        int i =0;
        int k =0;
        int j =size-1;
        while(i< size){
            
            if(nums[i] == 0)
            {
                ans[j] = 0;
                j--;
                
            }
            else{
                ans[k] = nums[i];
                k++;
            }
            
            i++;
            
        }
        
        nums = ans;
        
    }
};

File: 0283-move-zeroes/0283-move-zeroes.py
class Solution:
    def moveZeroes(self, nums: List[int]) -> None:
        """
        Do not return anything, modify nums in-place instead.
        """
        
        size = len(nums)
        i = 0
        j = 0
        
        while j<size:
            
            while j< size and nums[j] == 0:
                j+=1
                
            if j == size:
                break
            
            nums[i], nums[j] = nums[j], nums[i]
            i+=1
            j+=1
        